src/lib.rs

```diff
@@ -260,23 +287,54 @@ impl<'de> Deserialize<'de> for Message {
     }
 }
 
+#[cfg_attr(feature = "typed-builder", derive(typed_builder::TypedBuilder))]
+#[cfg_attr(feature = "typed-builder", builder(doc))]
+#[derive(Default)]
+/// Verification options and configuration
+pub struct VerificationOpts {
+    /// RFC 4501 dns authority that is requesting the signing (Optional)
+    pub domain: Option<Authority>,
+    /// Randomized token used to prevent replay attacks, at least 8 alphanumeric characters (Optional)
+    pub nonce: Option<String>,
+    /// ISO 8601 datetime string of the current time (Optional)
+    pub timestamp: Option<OffsetDateTime>,
```

Comment:
This might want to be changed to something like:
```suggestion
    /// ISO 8601 datetime string of the time that the message should be valid at (Optional)
    pub timestamp: Option<OffsetDateTime>,
```

Reply:
Gosh, they were all awful